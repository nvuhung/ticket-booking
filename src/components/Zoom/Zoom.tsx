import React from 'react';
import styled from 'styled-components';

import { theme } from '../../styles/theme';
import Icon from '../Icon';

export interface IProps {
  onZoomIn: () => void;
  onZoomOut: () => void;
}

const Zoom: React.FC<IProps> = ({ onZoomIn, onZoomOut }) => {
  return (
    <Container>
      <Button onClick={onZoomOut}>
        <Icon width={20} height={20} name="zoom-out" />
      </Button>
      <Button onClick={onZoomIn}>
        <Icon width={20} height={20} name="zoom-in" />
      </Button>
    </Container>
  );
};

const Container = styled.div`
  height: 60px;
  background-color: ${theme.colors.white};
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 26px;
  position: relative;
  outline: none;
  margin-right: 20px;
`;

const Button = styled.button`
  outline: none;
  cursor: pointer;
  background: transparent;
  display: flex;
  align-items: center;
  height: 30px;
  border: none;
  &:focus {
    outline: none;
  }
`;

export default React.memo(Zoom);
