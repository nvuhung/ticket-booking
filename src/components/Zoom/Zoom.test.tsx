import React from 'react';
import 'jest-styled-components';

import Zoom, { IProps } from './Zoom';
import { mount, shallow } from '../../test/enzyme';

describe('Zoom', () => {
  let props: IProps = (null as any) as IProps;
  let mountedComp: any;

  const wrapper = (bShallow: boolean) => {
    if (!mountedComp) {
      mountedComp = bShallow ? shallow(<Zoom {...props} />) : mount(<Zoom {...props} />);
    }
    return mountedComp;
  };

  beforeEach(() => {
    props = {
      // tslint:disable-next-line: no-empty
      onZoomIn: () => {},
      // tslint:disable-next-line: no-empty
      onZoomOut: () => {}
    };
  });

  it('renders without crashing', () => {
    const comp = wrapper(false);
    expect(comp).toMatchSnapshot();
  });

  it('always renders 2 buttons', () => {
    const comp = wrapper(false);
    expect(comp.find('button').length).toEqual(2);
  });
});
