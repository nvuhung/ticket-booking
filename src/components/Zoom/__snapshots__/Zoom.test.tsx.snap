// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zoom renders without crashing 1`] = `
.c0 {
  height: 60px;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 26px;
  position: relative;
  outline: none;
  margin-right: 20px;
}

.c1 {
  outline: none;
  cursor: pointer;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 30px;
  border: none;
}

.c1:focus {
  outline: none;
}

<Memo(Zoom)
  onZoomIn={[Function]}
  onZoomOut={[Function]}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  height: 60px;
  background-color: ",
              "#fff",
              ";
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 26px;
  position: relative;
  outline: none;
  margin-right: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styled.button
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  outline: none;
  cursor: pointer;
  background: transparent;
  display: flex;
  align-items: center;
  height: 30px;
  border: none;
  &:focus {
    outline: none;
  }
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="c1"
              onClick={[Function]}
            >
              <Icon
                height={20}
                name="zoom-out"
                width={20}
              >
                <ForwardRef(SvgIcZoomOut)
                  height={20}
                  width={20}
                >
                  <svg
                    height={20}
                    width={20}
                  >
                    ic-zoom-out.svg
                  </svg>
                </ForwardRef(SvgIcZoomOut)>
              </Icon>
            </button>
          </StyledComponent>
        </styled.button>
        <styled.button
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  outline: none;
  cursor: pointer;
  background: transparent;
  display: flex;
  align-items: center;
  height: 30px;
  border: none;
  &:focus {
    outline: none;
  }
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="c1"
              onClick={[Function]}
            >
              <Icon
                height={20}
                name="zoom-in"
                width={20}
              >
                <ForwardRef(SvgIcZoomIn)
                  height={20}
                  width={20}
                >
                  <svg
                    height={20}
                    width={20}
                  >
                    ic-zoom-in.svg
                  </svg>
                </ForwardRef(SvgIcZoomIn)>
              </Icon>
            </button>
          </StyledComponent>
        </styled.button>
      </div>
    </StyledComponent>
  </styled.div>
</Memo(Zoom)>
`;
